relay-system/src/controller.rs

```diff
@@ -110,6 +133,10 @@ impl Controller {
         // Send a shutdown signal to all registered subscribers (including self). They will report
         // when the shutdown has completed. Note that we ignore all errors to make sure that we
         // don't cancel the shutdown of other actors if one actor fails.
+
+        // Send the message
+        let _ = self.shutdown_sender.send(Some(Shutdown { timeout }));
```

Comment:
```suggestion
        self.shutdown_sender.send(Some(Shutdown { timeout }));
```

Reply:
If we remover the `let _ = `clippy will complain about:

> unused `Result` that must be used
> `#[warn(unused_must_use)]` on by default

Reply:
oh, right.  I guess we want to ignore errors if there are no receivers anymore.

I had no idea that doing `let _ =` is a workaround for that!  The common way (and I'd argue "right" way since it's more explicit) to do this is to turn the `Result` into an `Option` using `.ok()` and not assign that.

Reply:
```suggestion
        self.shutdown_sender.send(Some(Shutdown { timeout })).ok();
```

Reply:
Similar to actix, we could add a public function `fn do_send(&self, msg: M)` that is neither async nor returns anything. Inside, we can have this with a little comment explaining why.

Reply:
to be fair, if we're not saving anything, by e.g. introducing a `SendMessage<T>` so that the work of replying isn't done at all, then I prefer a smaller API with combinators (`.ok()`) as I think it's simpler and equally explicit.  But I don't feel that strongly.

Reply:
@flub you can't do `.ok()` here because that would require us to `.await.ok()`, and we do not want to await. Hence my suggestion for a `do_send` or similar method that can handle this internally in a non-blocking fashion.

Reply:
For what it is worth, I added the `.ok()` a couple of commits ago and clippy, cargo check and the tests are happy.

Reply:
Sorry about that, it took me until now that I confused it with [this](https://github.com/getsentry/relay/pull/1349/files#diff-cacfe5fee9e90beceb7a57ba096fbb7cdb4070e47a0d5a4444d8b68e8cbf0c42R155) instance where we call `Addr::send`, which is async.